cf-custom-resources/lib/custom-domain-app-runner.js

```diff
@@ -0,0 +1,189 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+/* jshint node: true */
+/*jshint esversion: 8 */
+
+"use strict";
+
+const AWS = require('aws-sdk');
+const { report, defaultSleep } = require('../lib/partials');
+
+const ERR_NAME_CUSTOM_DOMAIN_ALREADY_ASSOCIATED = "CustomDomainAlreadyAssociatedException";
+const ERR_NAME_INVALID_REQUEST = "InvalidRequestException";
+const DOMAIN_STATUS_PENDING_VERIFICATION = "pending_certificate_dns_validation";
+const ATTEMPTS = 10;
+
+
+let sleep = defaultSleep;
+let appRoute53Client, appRunnerClient, appHostedZoneID;
+
+exports.handler = async function (event, context) {
+    const props = event.ResourceProperties;
+    const [serviceARN, appDNSRole, customDomain] = [props.ServiceARN, props.AppDNSRole, props.CustomDomain,];
+    appHostedZoneID = props.HostedZoneID;
+
+    // Configure clients.
+    appRoute53Client = new AWS.Route53({
+        credentials: new AWS.ChainableTemporaryCredentials({
+            params: { RoleArn: appDNSRole, },
+            masterCredentials: new AWS.EnvironmentCredentials("AWS"),
+        }),
+    });
+    appRunnerClient = new AWS.AppRunner();
+
+    let addCustomDomainErr;
+    await addCustomDomain(serviceARN, customDomain).catch(async err => {
+        addCustomDomainErr = err;
+        if (err.name === "CustomDomainAlreadyAssociated") {
+            console.log("Custom domain already associated. Do nothing.");
+            return;
+        }
+        console.log(`Caught error: ${err.message}`);
+        await report(event, context, "FAILED", event.LogicalResourceId, null, err.message).catch((err) => {
+            throw new Error("send response: " + err.message);
+        });
+    });
+
+    if (!addCustomDomainErr) {
```

Comment:
If the failure `report` doesn't throw any error, it would execute the rest of the code (i.e. the success `report`)

Reply:
yeah but even if it throws error we could still do the success `report` right? since it would be the associated do nothing error.

Reply:
Actually it is the `CustomDomainAlreadyAssociated` error, it will directly go to the success `report`. There is a `return` statement if `CustomDomainAlreadyAssociated` is detected.

The failure report happens for all errors that're not `CustomDomainAlreadyAssociated` (i.e. real failure)